internal/infrastructure/elasticsearch/elastic_search_service.go

```diff
@@ -101,9 +102,13 @@ func (d *ElasticSearch) buildAndExecRequest(method, url string, obj interface{},
 		time.Sleep(backoff)
 	}
 	if objOut != nil {
-		err = json.NewDecoder(resp.Body).Decode(&objOut)
-		if err != nil {
-			return nil, errors.Wrap(err, "error decoding response json")
+		if resp != nil {
+			err = json.NewDecoder(resp.Body).Decode(&objOut)
```

Comment:
Hm, is it the caller's responsibility to close the body? It doesn't seem like any of the callers are.